cpp/src/runtime/context.cpp

```diff
@@ -26,13 +27,23 @@
 
 namespace cudf {
 
-context::context() : _program_cache{std::make_unique<jit::program_cache>()}
+context::context(init_flags flags) : _program_cache{nullptr}
 {
+  if (has_flag(flags, init_flags::INIT_JIT_CACHE)) {
+    _program_cache = std::make_unique<jit::program_cache>();
+  }
+
+  if (has_flag(flags, init_flags::LOAD_NVCOMP)) { io::detail::nvcomp::load_nvcomp_library(); }
+
   auto dump_codegen_flag = getenv_or("LIBCUDF_JIT_DUMP_CODEGEN", std::string{"OFF"});
   _dump_codegen          = (dump_codegen_flag == "ON" || dump_codegen_flag == "1");
 }
 
-jit::program_cache& context::program_cache() { return *_program_cache; }
+jit::program_cache& context::program_cache()
+{
+  CUDF_EXPECTS(_program_cache != nullptr, "JIT cache not initialized", std::runtime_error);
```

Comment:
Just validating: in practice this is unreachable because we only construct a single context object as a singleton via `get_context`, right? In order to hit this the our jit::get_program_cache function would have to be modified to use a different context?

Reply:
this is here for the case when the singleton is initialized without the INIT_JIT_CACHE flag.